java/compaction/compaction-status-store/src/main/java/sleeper/compaction/status/store/job/DynamoDBCompactionJobStatusStore.java

```diff
@@ -48,6 +47,7 @@
 import static sleeper.configuration.properties.UserDefinedInstanceProperty.COMPACTION_STATUS_STORE_ENABLED;
 import static sleeper.configuration.properties.UserDefinedInstanceProperty.ID;
 import static sleeper.dynamodb.tools.DynamoDBAttributes.createStringAttribute;
+import static sleeper.dynamodb.tools.DynamoDBUtils.doScanWithPagination;
 import static sleeper.dynamodb.tools.DynamoDBUtils.instanceTableName;
 
 public class DynamoDBCompactionJobStatusStore implements CompactionJobStatusStore {
```

Comment:
These changes could be tested by controlling the scan limit in tests. We could leave it as null in production, but set it to a very low value for a test, to reproduce the problem and to ensure that the pagination works correctly.